clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.clustering.marshalling.protostream;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.OptionalInt;
+import java.util.function.Supplier;
+
+import org.infinispan.protostream.BaseMarshaller;
+import org.infinispan.protostream.ImmutableSerializationContext;
+
+import protostream.com.google.protobuf.CodedOutputStream;
+
+/**
+ * {@link ScalarProtoStreamWriter} implementation that does not write to any stream, but instead computes the number of bytes that would be written to a stream.
+ * @author Paul Ferraro
+ */
+public class SizeComputingProtoStreamWriter implements ProtoStreamWriter, Supplier<OptionalInt> {
+
+    private final ImmutableSerializationContext context;
+    private int size = 0;
+    private boolean present = true;
```

Comment:
@pferraro I can't see any place where this is set to any value but 'true'.

Reply:
https://github.com/pferraro/wildfly/blob/WFLY-14359/clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java#L235

Reply:
Thanks